api/src/schema/resolvers.js

```diff
@@ -31,7 +41,40 @@ const resolvers = {
         )
       }
 
-      let query = createQuery(forwardSortationArea, filter)
+      let query = {
+        $and: [
+          {
+            forwardSortationArea,
+          },
+        ],
+      }
+
+      if (filter) {
+        for (let key in filter) {
+          // make sure the key exists in the comparators' keys
+          if (Object.keys(comparators).includes(key)) {
+            // if the value is a number
+            // - optionally starting with a minus sign
+            // - containing zero or one decimal places
+            // convert it to a float instead of a string
+            if (filter[key].match(/^-?\d+\.?\d+$/)) {
+              filter[key] = parseFloat(filter[key])
+            }
+
+            // We are evaling our own code here, not user input.
+            // Filter.field is our code, stringified and stored in an enum.
+            // The user choses one of the enum values and we convert the string
+            // back to a function which accepts a matcher and generates the
+            // query we need to find that field.
+            // eslint-disable-next-line no-eval
+            let queryGenerator = eval(filter.field)
+            let attrQuery = queryGenerator({
+              [comparators[key]]: filter[key],
+            })
+            query['$and'].push(attrQuery)
```

Comment:
Looks really clean though, eh? I know we normally frown on `eval`s, but I think this is pretty neat.

Reply:
You can see how all of a sudden multiple queries are going to be easy. :smile: